integration_tests/steps/area_events.py

```diff
@@ -153,3 +156,64 @@ def load_does_not_consume_between(context, start_time, end_time):
         if not start_time <= market.time_slot.hour < end_time:
             continue
         assert len(market.trades) == 0
+
+
+def _read_solar_profile(profile_path, slot_length):
+    return read_profile_csv_to_dict(InputProfileTypes.POWER, str(profile_path), slot_length)
+
+
+@then('both PVs follow they sunny profile until 12:00')
+def both_sunny_profile(context):
+    slot_length = context.simulation.area.config.slot_length
+    sunny = _read_solar_profile(d3a_path + '/resources/Solar_Curve_W_sunny.csv', slot_length)
+    house1 = next(filter(lambda x: x.name == "House 1", context.simulation.area.children))
+    house2 = next(filter(lambda x: x.name == "House 2", context.simulation.area.children))
+
+    for time_slot, market in house1._markets.past_markets.items():
+        if time_slot.hour > 12:
+            continue
+        h1_trades = [t for t in market.trades if t.seller == "H1 PV"]
+        h2_trades = [t for t in house2._markets.past_markets[time_slot].trades
```

Comment:
Why are the h2_trades determined different from the h1_trades?

Reply:
This is necessary in order to get the trades that happen in the same timeslot in 2 different IAAs. The rationale behind this is to iterate over all markets of house 1, and get the trades that happened in house 2 at this market slot in order to see at this exact point in time what happened to all the IAAs.

Reply:
Ahh, now i get it. `market` is the house_1 market. 
Ok, Thanks!